sa/sa.go

```diff
@@ -1375,11 +1378,38 @@ func (ssa *SQLStorageAuthority) NewAuthorizations2(ctx context.Context, req *sap
 		if err != nil {
 			return nil, err
 		}
-		err = ssa.dbMap.Insert(am)
+
+		// Each authz needs a (?,?...), in the VALUES block
+		qs.WriteString(fmt.Sprintf("(%s),", strings.TrimRight(strings.Repeat(" ?,", 11), ",")))
+		// Must follow the order of authzFields
+		vals = append(vals, am.ID)
+		vals = append(vals, am.IdentifierType)
+		vals = append(vals, am.IdentifierValue)
+		vals = append(vals, am.RegistrationID)
+		vals = append(vals, am.Status)
+		vals = append(vals, am.Expires)
+		vals = append(vals, am.Challenges)
+		vals = append(vals, am.Attempted)
+		// AttemptedAt is not included in the authzFields
+		vals = append(vals, am.Token)
+		vals = append(vals, am.ValidationError)
+		vals = append(vals, am.ValidationRecord)
+	}
```

Comment:
- The `append()`'s second argument is a variadic; we can append all of these values at once, in order
- Nit, these longer comments should be complete sentences ending in punctuation
- Nit, for consistency and readability we should insert a blank line before comments

```suggestion

		// Append the values following the order of `AuthzFields`.
		queryArgs = append(vals, am.ID,
			am.IdentifierType,
			am.IdentifierValue,
			am.RegistrationID,
			am.Status,
			am.Expires,
			am.Challenges,
			am.Attempted,
			am.Token,
			am.ValidationError,
			am.ValidationRecord,
		)
	}

```

Reply:
OK